### PR TITLE
 📖 Fix target management cluster broken link

### LINKED ISSUE #2072: Fix glossary terminology around management clusters
/kind bug
/area documentation

https://github.com/kubernetes-sigs/cluster-api/pull/2042, fixes a linking error, but we have confusing terminology around "target management cluster", which dates to when clusterctl v1 pivoting was the predominant way of using Cluster API.

This terminology needs to be cleaned up in relation to management and guest clusters.

### PR DESCRIPTION
**Pull request by outofmem0ry**
**What this PR does / why we need it**:
Changes target cluster to target management cluster as defined in the glossary. Also, fixes the broken link to redirect to the correct definition in the glossary. 

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #2072

### CODE REVIEW

**Comment by vincepri** [comment]
Hmm, I think we should revisit the glossary a little, `target management cluster` is a weird term imo, which suggests that the target cluster is also a management cluster. We usually refer to `target cluster` as a cluster managed by Cluster API, regardless where the management cluster lives. I think the text is actually correct, but the glossary needs to be changed.

**Comment by randomvariable** [comment]
Agreed. @dlipovetsky and I were both a bit puzzled about the terminology. Looks like it was written when the pivot workflow was predominant, and we don't have corresponding text for the non-pivot workflow.

**Comment by randomvariable** [comment]
Going to approve this, as it fixes the immediate problem, but filed https://github.com/kubernetes-sigs/cluster-api/issues/2072 as a new issue.

/approve
/lgtm

File: docs/book/src/user/quick-start.md [hunk 1]
```diff
@@ -731,7 +731,7 @@ spec:
 {{#/tab }}
 {{#/tabs }}
 
-After the controlplane is up and running, let's retrieve the [target cluster] Kubeconfig:
+After the controlplane is up and running, let's retrieve the [target management cluster] Kubeconfig:
```
Comment (chuckha) [inline docs/book/src/user/quick-start.md]: This is actually the workload cluster. But I see Workload cluster is missing from the glossary.

If you change this to 
```suggestion
After the controlplane is up and running, let's retrieve the [workload cluster] Kubeconfig:
```

And add the Workload cluster definition to the glossary and link to it then we'll be good to get this merged

**Comment by vincepri** [comment]
/hold 

I would rather fix everything in this PR per @chuckha's suggestions

**Comment by wfernandes** [comment]
Not to add to the confusion but after some quick reading I came across the following:
**Management Cluster**
This term is mentioned in the [Concepts section](https://master.cluster-api.sigs.k8s.io/user/concepts.html#management-cluster) and in the [Glossary](https://master.cluster-api.sigs.k8s.io/reference/glossary.html#management-cluster).
**Workload/Target Cluster**
This term is also mentioned in the [Concepts section](https://master.cluster-api.sigs.k8s.io/user/concepts.html#workloadtarget-cluster) and in the [Glossary](https://master.cluster-api.sigs.k8s.io/reference/glossary.html#workload-cluster).

So I think it would make sense to just remove the **Target management cluster** term from the gloassary.

**Comment by detiber** [comment]
So, originally Target was used to differentiate pivot terminology from standard management terminology (management/workload).

As long as we have some concept of a pivot, I think we should keep the differentiating terminology.

Management terminology:
- Management Cluster: A Kubernetes Cluster running the Cluster API components that manages one or more Workload Clusters
- Workload Cluster: A Kubernetes Cluster managed by a Management Cluster

Pivot Terminology:
- Source Cluster: A Kubernetes Cluster running the Cluster API components that will be migrated to the Target Cluster
- Target Cluster: A Kubernetes Cluster that will be running the Cluster API components that will be migrated from the Source Cluster

**Comment by ncdc** [comment]
@outofmem0ry hi, checking in! Do you think you'll be able to make the requested changes soon?

**Comment by outofmem0ry** [comment]
@ncdc planning to wrap this up today.

**Comment by ncdc** [comment]
@outofmem0ry checking in again - do you have any updates? Thanks!

**Pull request merged**